src/cliques.c

```diff
@@ -0,0 +1,628 @@
+  }
+  if (hook_obj == Fail) {
+    if (!IS_LIST(user_param_obj) || !IS_MUTABLE_OBJ(user_param_obj)) {
+      ErrorQuit("the 2rd argument <hook> is fail and so the 3th argument must "
+                "be a mutable list, not %s,",
+                (Int) TNAM_OBJ(user_param_obj),
+                0L);
+    }
+  } else if (!IS_FUNC(hook_obj) || NARG_FUNC(hook_obj) != 2) {
+    ErrorQuit(
+        "the 2rd argument <hook> must be a function with 2 arguments,", 0L, 0L);
+  }
+  if (!IS_INTOBJ(limit_obj) && limit_obj != Infinity) {
+    ErrorQuit("the 4th argument <limit> must be an integer "
+              "or infinity, not %s,",
+              (Int) TNAM_OBJ(limit_obj),
+              0L);
+  } else if (IS_INTOBJ(limit_obj) && INT_INTOBJ(limit_obj) <= 0) {
+    ErrorQuit("the 4th argument <limit> must be a positive integer, "
+              "not %d,",
+              INT_INTOBJ(limit_obj),
+              0L);
+  }
+  if (!IS_LIST(include_obj) && include_obj != Fail) {
+    ErrorQuit("the 5th argument <include> must be a list or fail, not %s,",
+              (Int) TNAM_OBJ(include_obj),
+              0L);
+  } else if (IS_LIST(include_obj)) {
+    for (Int i = 1; i <= LEN_LIST(include_obj); ++i) {
+      if (!ISB_LIST(include_obj, i)) {
+        ErrorQuit("the 5th argument <include> must be a dense list,", 0L, 0L);
+      } else if (!IS_POS_INT(ELM_LIST(include_obj, i))) {
+        ErrorQuit("the 5th argument <include> must only contain positive "
+                  "integers, but found %s in position %d,",
+                  (Int) TNAM_OBJ(ELM_LIST(include_obj, i)),
+                  i);
+      } else if (INT_INTOBJ(ELM_LIST(include_obj, i)) 
+                 > DigraphNrVertices(digraph_obj)) {
+        ErrorQuit("in the 5th argument <include> position %d is out of range, "
+                  "must be in the range [1, %d],",
+                  i,
+                  DigraphNrVertices(digraph_obj));
+      } else if (INT_INTOBJ(
+                     POS_LIST(include_obj, ELM_LIST(include_obj, i), INTOBJ_INT(0)))
+                 < i) {
+        ErrorQuit("in the 5th argument <include> position %d is a duplicate,", i, 0L);
+      }
+    }
+  }
+  if (!IS_LIST(exclude_obj) && exclude_obj != Fail) {
+    ErrorQuit("the 6th argument <exclude> must be a list or fail, not %s,",
+              (Int) TNAM_OBJ(exclude_obj),
+              0L);
+  } else if (IS_LIST(exclude_obj)) {
+    for (Int i = 1; i <= LEN_LIST(exclude_obj); ++i) {
+      if (!ISB_LIST(exclude_obj, i)) {
+        ErrorQuit("the 6th argument <exclude> must be a dense list,", 0L, 0L);
+      } else if (!IS_POS_INT(ELM_LIST(exclude_obj, i))) {
+        ErrorQuit("the 6th argument <exclude> must only contain positive "
+                  "integers, but found %s in position %d,",
+                  (Int) TNAM_OBJ(ELM_LIST(exclude_obj, i)),
+                  i);
+      } else if (INT_INTOBJ(ELM_LIST(exclude_obj, i)) 
+                 > DigraphNrVertices(digraph_obj)) {
```

Comment:
This is not safe: `IS_POS_INT` will return true also for values like `2^1000` which are not intobj; and then the above command can either accidentally accept an invalid value (although that's unlikely), or crash (if in debug mode).

You could either use `IS_POS_INTOBJ` above (but then also adjust the error message, e.g. to say `... positive small integers` (the term "small integers" is defined precisely in the GAP manual. Well, at least since GAP 4.11 it is... ). Or else adjust the check here, e.g. by inserting an `IS_INTOBJ` or `IS_POS_INTOBJ` call before the `INT_INTOBJ`.